helpers/wharfapiext/extension.go

```diff
@@ -0,0 +1,181 @@
+package wharfapiext
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"reflect"
+	"regexp"
+
+	"github.com/iver-wharf/wharf-api-client-go/pkg/wharfapi"
+	log "github.com/sirupsen/logrus"
+)
+
+// ExtClient Extension of the wharfapi.Client
+type ExtClient struct {
+	*wharfapi.Client
+}
```

Comment:
We do control the `wharfapi` library, so I suggest moving your code over there, in as appropriate way as you can imagine without draggin the AzureDevOps domain into it.

Looks like you've copy-pasted a lot of code from the `wharfapi` library itself. Even introduced logrus from there.

Sorry, but I suggest rethinking this `wharfapiext` package. :(

Reply:
I don't quite remember why I even created it, so that's a fair suggestion.
It doesn't seem to be anything other than some refactoring of existing functions.